thorlcr/graph/thgraph.cpp

```diff
@@ -320,7 +322,47 @@ CGraphElementBase *createGraphElement(IPropertyTree &node, CGraphBase &owner, CG
     return container;
 }
 
-CGraphElementBase::CGraphElementBase(CGraphBase &_owner, IPropertyTree &_xgmml) : owner(&_owner)
+///////////////////////////////////
+CActivityCodeContext::CActivityCodeContext()
+{
+}
+
+IThorChildGraph * CActivityCodeContext::resolveChildQuery(__int64 gid, IHThorArg * colocal)
+{
+    return parent->getChildGraph((graph_id)gid);
+}
+
+IEclGraphResults * CActivityCodeContext::resolveLocalQuery(__int64 gid)
+{
+    if (gid == containerGraph->queryGraphId())
+        return containerGraph;
+    else
+        return ctx->resolveLocalQuery(gid);
+}
+
+unsigned CActivityCodeContext::getGraphLoopCounter() const
+{
+    return containerGraph->queryLoopCounter();           // only called if value is valid
+}
+
+ISectionTimer * CActivityCodeContext::registerTimer(unsigned activityId, const char * name)
+{
+    if (!stats) // if master context and local CQ, there is no activity instance, and hence no setStats call
+        return queryNullSectionTimer();
+    CriticalBlock b(contextCrit);
+    auto it = functionTimers.find(name);
```

Comment:
This isn't particularly time criticial, so it doesn't matter.  However do you know if this will cause a temporary std::string to be created?

Reply:
I think the answer is yes.  Something to bear in mind for cases that are time critical.